web_programming/fetch_well_rx_price.py

```diff
@@ -0,0 +1,123 @@
+"""
+
+Scrape the price and pharmacy name for a prescription drug from rx site
+after providing the drug name and zipcode. 
+
+"""
+
+import lxml
+
+from typing import Union
+from requests import Response, get
+from bs4 import BeautifulSoup
+
+
+def format_price(price: str) -> float:
+    """[summary]
+
+    Remove the dollar from the string and convert it to float.
+
+    Args:
+        price (str): [price of drug in string format]
+
+    Returns:
+        float: [formatted price of drug in float]
+
+    >>> format_price("$14")
+    14.0
+
+    >>> format_price("$15.67")
+    15.67
+
+    >>> format_price("$0.00")
+    0.0
+
+    """
+    dollar_removed: str = price.replace("$", "")
+    formatted_price: float = float(dollar_removed)
+    return formatted_price
+
+
+def fetch_pharmacy_and_price_list(drug_name: str, zip_code: str) -> Union[list, None]:
+    """[summary]
+
+    This function will take input of drug name and zipcode, then request to the BASE_URL site,
+    Get the page data and scrape it to the generate the list of lowest prices for the prescription drug.
+
+    Args:
+        drug_name (str): [Drug name]
+        zip_code(str): [Zip code]
+
+    Returns:
+        list: [List of pharmacy name and price]
+
+    >>> fetch_pharmacy_and_price_list(None, None)
+
+    >>> fetch_pharmacy_and_price_list(None, 30303)
+
+    >>> fetch_pharmacy_and_price_list("eliquis", None)
+
+    """
+
+    try:
+
+        # Has user provided both inputs?
+        if not drug_name or not zip_code:
+            return None
+
+        request_url: str = f'https://www.wellrx.com/prescriptions/{drug_name}/{zip_code}/?freshSearch=true'
+        response: Response = get(request_url)
```

Comment:
Let's put type hints on function parameters and function return types but we do not need them _everywhere_.  Both Python and mypy are capable of figuring out that a string literal is a string.  ;-)  Overuse slows down both the writer and reader of code.
```suggestion
        request_url = f'https://www.wellrx.com/prescriptions/{drug_name}/{zip_code}/?freshSearch=true'
        response = get(request_url)
```